incident_viewer_demo/src/LineChart/LineChart.jsx

```diff
@@ -305,37 +316,86 @@ class LineChart extends PureComponent {
       const badYmax = yScale(badIncidentSeries.download_speed_mbps_median);
       this.infoHoverBox.selectAll('*').remove();
 
-      // TODO: Change values to use the correct JSON attributes
-
-      // Draw the hover state for the good period information
-      if (highlightedDate.isSameOrBefore(goodIncidentSeries.end) && highlightedDate.isSameOrAfter(goodIncidentSeries.start) && mouseY > goodYmax) {
-        
-        this.infoHoverBox.append('rect')
-        .classed("good-incident-area", true)
-        .attr('x', xScale(goodIncidentSeries.start))
-        .attr('y', goodYmax)
-        .attr('width', xScale(goodIncidentSeries.end) - xScale(goodIncidentSeries.start))
-        .attr('height', plotAreaHeight-goodYmax)
-      }
-
-      // Draw the hover state for the bad period information
-      if (highlightedDate.isSameOrBefore(badIncidentSeries.end) && highlightedDate.isSameOrAfter(badIncidentSeries.start) && mouseY > badYmax) {
-        this.infoHoverBox.append('rect')
-        .classed("bad-incident-area", true)
-        .attr('x', xScale(badIncidentSeries.start))
-        .attr('y', badYmax)
-        .attr('width', xScale(badIncidentSeries.end) - xScale(badIncidentSeries.start))
-        .attr('height', plotAreaHeight-badYmax)
-      }
-
-      // Draw the hover state for the incident information
-      if (highlightedDate.isSameOrBefore(badIncidentSeries.end) && highlightedDate.isSameOrAfter(badIncidentSeries.start) && mouseY < badYmax && mouseY > goodYmax) {
-        this.infoHoverBox.append('rect')
-        .classed("incident-area", true)
-        .attr('x', xScale(badIncidentSeries.start))
-        .attr('y', goodYmax)
-        .attr('width', xScale(badIncidentSeries.end) - xScale(badIncidentSeries.start))
-        .attr('height', badYmax-goodYmax)
+      if (hasIncident) {
+        // TODO: Change values to use the correct JSON attributes
+        const width = xScale(goodIncidentSeries.end) - xScale(goodIncidentSeries.start);
+        const rectFitsText = width > 180;
+        const verticalTextShifter = 0.45; // NOTE: This needs to be tuned based on font size and number of lines :(
+
+        // Draw the hover state for the good period information
+        if (highlightedDate.isBefore(goodIncidentSeries.end) && highlightedDate.isSameOrAfter(goodIncidentSeries.start) && mouseY > goodYmax) {
+          this.infoHoverBox.append('rect')
+          .classed("good-incident-area", true)
+          .attr('x', xScale(goodIncidentSeries.start))
+          .attr('y', goodYmax)
+          .attr('width', width)
+          .attr('height', plotAreaHeight-goodYmax);
+
+          if (rectFitsText) {
```

Comment:
Please add a comment here indicating that if the area cannot fit the text, then no text will be displayed on the viz.